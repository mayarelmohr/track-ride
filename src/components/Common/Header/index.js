import React from "react";
import styles from "./styles.module.css";

const Header = React.memo(() => {
  return (
    <header className={styles.header}>
      <h1 className={styles.wrapper}>
        <span className={styles.hidden}>Swvl</span>
        <img
          className={styles.logo}
          src="https://swvl.com/assets/img/swvl-new-logo.png"
          alt="swvl"
        />
      </h1>
    </header>
  );
});

export default Header;
