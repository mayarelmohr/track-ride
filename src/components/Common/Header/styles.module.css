.hidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.logo {
  width: 90px;
  vertical-align: middle;
}

.header {
  padding: 10px 30px;
  height: 60px;
  line-height: 60px;
  box-shadow: 7px 7px 8px -8px #b4b4b4;
  position: relative;
  z-index: 1;
}

.wrapper {
  margin: 0;
}
