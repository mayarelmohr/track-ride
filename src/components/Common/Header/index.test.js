import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import Header from "./index";

const headerElement = (
  <MemoryRouter>
    <Header />
  </MemoryRouter>
);

test("header renders correctly", () => {
  const { getByAltText } = render(headerElement);
  const logo = getByAltText("swvl");
  expect(logo).toBeTruthy();
});

test("it renders ", () => {
  const { asFragment } = render(headerElement);
  expect(asFragment()).toMatchSnapshot();
});
