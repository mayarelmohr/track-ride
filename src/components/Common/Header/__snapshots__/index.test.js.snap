// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders  1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <a
      data-testid="home"
      href="/"
    >
      <h1
        class="wrapper"
      >
        <span
          class="hidden"
        >
          Swvl
        </span>
        <img
          alt="swvl"
          class="logo"
          src="https://swvl.com/assets/img/swvl-new-logo.png"
        />
      </h1>
    </a>
  </header>
</DocumentFragment>
`;
